src/bin/rustfmt.rs

```diff
@@ -306,8 +321,8 @@ fn determine_operation(matches: &Matches) -> FmtResult<Operation> {
             Some(dir)
         });
 
-    // if no file argument is supplied, read from stdin
-    if matches.free.is_empty() {
+    // if no file argument is supplied and `--file-lines` is not specified, read from stdin
+    if matches.free.is_empty() && !matches.opt_present("file-lines") {
```

Comment:
Accepting `file: <stdin>` sounds good. I would allow --file-lines and a file specified on the command line, as long as the names match up - seems silly to do, but no reason to error out. You can leave all of this till later though.